Source/Timeline/EventView.swift

```diff
@@ -0,0 +1,88 @@
+import UIKit
+import DateTools
+import Neon
+import DynamicColor
+
+public class EventView: UIView {
+
+  var color = UIColor() {
+    didSet {
+      textView.textColor = color.darkenColor(0.3)
+      backgroundColor = UIColor(red: color.redComponent(), green: color.greenComponent(), blue: color.blueComponent(), alpha: 0.3)
+    }
+  }
+
+  var contentHeight: CGFloat {
+    //TODO: use strings array to calculate height
+    return textView.height
+  }
+
+  var data = [String]() {
+    didSet {
+      textView.text = data.reduce("", combine: {$0 + $1 + "\n"})
+    }
+  }
+
+  lazy var textView: UITextView = {
+    let view = UITextView()
+    view.font = UIFont.boldSystemFontOfSize(12)
+    view.userInteractionEnabled = false
+    view.backgroundColor = UIColor.clearColor()
+
+    return view
+  }()
+
+  lazy var tapGestureRecognizer: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: "tap")
+  lazy var longPressGestureRecognizer: UILongPressGestureRecognizer = UILongPressGestureRecognizer(target: self, action: "longPress")
+
+  var datePeriod = DTTimePeriod()
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    configure()
+  }
+
+  required public init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+    configure()
+  }
+
+  func configure() {
+    clipsToBounds = true
+    [tapGestureRecognizer, longPressGestureRecognizer].forEach {addGestureRecognizer($0)}
+
+    color = tintColor
+    addSubview(textView)
+  }
+
+  func tap() {
+
```

Comment:
This part of the interface still is to be decided.
Right now it is for subclassing and implementing own tap handler. So without this method it will crash because of `unrecognized selector sent to instance`.

Another option is by using delegate pattern and provide delegate calls in those stubs.